node_modules/oae-principals/tests/test-emails.js

```diff
@@ -0,0 +1,603 @@
+                    assert.ok(!err);
+                    assert.strictEqual(me.emailVerified, true);
+
+                    // Sanity-check there's a mapping for it
+                    PrincipalsTestUtil.assertUserEmailMapping(me.email, [me.id], function() {
+
+                        // Update the email address
+                        var email = TestsUtil.generateTestEmailAddress();
+                        PrincipalsTestUtil.assertUpdateUserSucceeds(simong.restContext, simong.user.id, {'email': email}, function(user, token) {
+
+                            // Assert the old mapping is still in place
+                            PrincipalsTestUtil.assertUserEmailMapping(oldEmailAddress, [me.id], function() {
+
+                                // Assert there's no mapping for the new email address as it hasn't been verified yet
+                                PrincipalsTestUtil.assertUserEmailMapping(email, [], function() {
+
+                                    RestAPI.User.getMe(simong.restContext, function(err, me) {
+                                        assert.ok(!err);
+
+                                        // The old email address should still be in place as the new one hasn't
+                                        // been verified yet
+                                        assert.strictEqual(me.email, oldEmailAddress);
+
+                                        // Assert the desired email address is recorded but not verified yet
+                                        assert.strictEqual(me.emailChange, email.toLowerCase());
+                                        assert.strictEqual(me.emailVerified, false);
+
+                                        // Assert we can verify the email address
+                                        PrincipalsTestUtil.assertVerifyEmailSucceeds(anonymousRestContext, email, token, function() {
+
+                                            // Assert the old mapping is gone
+                                            PrincipalsTestUtil.assertUserEmailMapping(oldEmailAddress, [], function() {
+
+                                                // Assert the new mapping is there
+                                                PrincipalsTestUtil.assertUserEmailMapping(email, [me.id], function() {
+                                                    return callback();
+                                                });
+                                            });
+                                        });
+                                    });
+                                });
+                            });
+                        });
+                    });
+                });
+            });
+        });
+
+        /**
+         * Test that verifies that users who are reverting to their old email address do not need to verify it
+         */
+        it('verify users reverting to their old email address do not need to verify it', function(callback) {
```

Comment:
We aren't "reverting" an email address so much as we are cancelling an email update verification. If we get rid of the `emailChange` and `emailVerified` properties, then this is probably a special action to "delete email verifications" for a user.